doc/api/util.md

```diff
@@ -398,6 +398,63 @@ util.inspect.defaultOptions.maxArrayLength = null;
 console.log(arr); // logs the full array
 ```
 
+## util.promisify(function)
+<!-- YAML
+added: REPLACEME
+-->
+
+* `function` {Function}
+
+Takes a function following the common Node.js callback style, i.e. taking a
+`(err, value) => ...` callback as the last argument, and returns a version
+that returns promises.
+
+For example:
+
+```js
+const util = require('util');
+const fs = require('fs');
+
+const stat = util.promisify(fs.stat);
+stat('.').then((stats) => {
+  // Do something with `stats`
+}).catch((error) => {
+  // Handle the error.
+});
+```
+
+If there is a `function[util.promisify.custom]` property present, `promisify`
+will return its value, see [Custom promisified functions][].
+
+### Custom promisified functions
+
+Using the `util.promisify.custom` one can override the return value of
+[`util.promisify()`][]:
+
+```js
+const util = require('util');
+
+function doSomething(foo, callback) {
+  // ...
+}
+
+doSomething[util.promisify.custom] = function(foo) {
+  return getPromiseSomehow();
+}
+
+const promisified = util.promisify(doSomething);
+console.log(promisified === doSomething[util.promisify.custom]);
+  // prints 'true'
+```
+
+### util.promisify.custom
+<!-- YAML
+added: REPLACEME
+-->
+
+A Symbol that can be used to declare custom promisified variants of functions,
+see [Custom promisified functions][].
```

Comment:
I think the uppercase variant is okay here, it’s referring to a literal section title, and it matches what we do for the other two section references in this file. If you feel strongly about it I’ll change it.